src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -514,6 +529,31 @@ private int WriteRunToErrorList(Run run, string logFilePath)
             return sarifErrors.Count;
         }
 
+        private void ShowFilteredSuppressionStateColumn()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            // Creating this table source adds "Suppression State" to the list of available columns.
+            SarifSuppressedResultsTableDataSource _ = SarifSuppressedResultsTableDataSource.Instance;
+
+            this.columnFilterer.FilterOut(
+                columnName: SarifResultTableEntry.SuppressionStateColumnName,
+                filteredValue: VSSuppressionState.Suppressed.ToString());
+        }
+
+        private void ShowFilteredCategoryColumn()
```

Comment:
>ShowFilteredCategoryColumn [](start = 21, length = 26)

same note as previous, add detailed comment and verify that we're not overwriting explicitly set user filters that make 'absent' results results visible.

Also, separately, we need an issue tracking our permanent mechanism for handling baseline state. It's not clear that repurposing the 'Category' column is the right thing to do for the long-term.

Reply:
The user's filter setting to display absent or suppressed results persists through the lifetime of the VS session. I've verified that repeatedly, and am in the process of enhancing the UI tests to demonstrate it. I filed [https://github.com/microsoft/sarif-visualstudio-extension/issues/315](https://github.com/microsoft/sarif-visualstudio-extension/issues/315), "Decide where to display Baseline State".

---
In reply to: [538747072](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538747072) [](ancestors = 538747072)